packages/features/pbac/services/permission-diff.service.ts

```diff
@@ -0,0 +1,48 @@
+import type { RolePermission } from "../domain/types/role";
+
+export type PermissionString = string;
+export type ParsedPermission = { resource: string; action: string };
+
+export class PermissionDiffService {
+  private parsePermission(permission: PermissionString): ParsedPermission {
+    const [resource, action] = permission.split(".");
+    return { resource, action };
+  }
+
+  private filterInternalPermissions(permissions: PermissionString[]): PermissionString[] {
+    return permissions.filter((permission) => {
+      const { action } = this.parsePermission(permission);
+      return action !== "_resource";
+    });
+  }
+
+  private createPermissionKey(permission: ParsedPermission): string {
+    return `${permission.resource}:${permission.action}`;
+  }
+
+  calculateDiff(newPermissions: PermissionString[], existingPermissions: RolePermission[]) {
```

Comment:
Function to take newPermissions coming form the UI and existing permissions this calculates what needs deleted and removed